File: Timed-Math-Challenge.go
package main

import (
    "fmt"
    "math/rand"
    "strconv"
    "time"
)

var (
    operators    = []string{"+", "-", "*"}
    minOperand   = 4
    maxOperand   = 15
    totalProblems = 10
)

func generateProblem() (string, int) {
    left := rand.Intn(maxOperand-minOperand+1) + minOperand
    right := rand.Intn(maxOperand-minOperand+1) + minOperand
    operator := operators[rand.Intn(len(operators))]

    expr := fmt.Sprintf("%d %s %d", left, operator, right)
    var answer int

    switch operator {
    case "+":
        answer = left + right
    case "-":
        answer = left - right
    case "*":
        answer = left * right
    }

    return expr, answer
}

func main() {
    rand.Seed(time.Now().UnixNano())

    var wrong int
    fmt.Println("Press Enter to start")
    fmt.Scanln()
    fmt.Println("--------------------------------")

    startTime := time.Now()

    for i := 0; i < totalProblems; i++ {
        expr, answer := generateProblem()
        for {
            fmt.Printf("Problem #%d: %s = ", i+1, expr)
            var guess string
            fmt.Scan(&guess)
            if guessInt, err := strconv.Atoi(guess); err == nil && guessInt == answer {
                break
            }
            wrong++
        }
    }

    totalTime := time.Since(startTime).Seconds()

    fmt.Println("--------------------------------")
    fmt.Printf("Nice Work! You finished in %.2f seconds!\n", totalTime)
}

File: Timed-Math-Challenge.js
const readline = require("readline");

const OPERATORS = ["+", "-", "*"];
const MIN_OPERAND = 4;
const MAX_OPERAND = 5;
const TOTAL_PROBLEMS = 5;

function generateProblem() {
  const left =
    Math.floor(Math.random() * (MAX_OPERAND - MIN_OPERAND + 1)) + MIN_OPERAND;
  const right =
    Math.floor(Math.random() * (MAX_OPERAND - MIN_OPERAND + 1)) + MIN_OPERAND;
  const operator = OPERATORS[Math.floor(Math.random() * OPERATORS.length)];
  const expr = `${left} ${operator} ${right}`;
  const answer = eval(expr);
  return { expr, answer };
}

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let wrong = 0;
console.log("Press any key to start");
rl.question("", (start) => {
  console.log("--------------------------------");
  const startTime = Date.now();

  let i = 0;
  function askQuestion() {
    if (i < TOTAL_PROBLEMS) {
      const { expr, answer } = generateProblem();
      rl.question(`Problem #${i + 1}: ${expr} = `, (guess) => {
        if (guess === answer.toString()) {
          i++;
          askQuestion();
        } else {
          wrong++;
          askQuestion();
        }
      });
    } else {
      const endTime = Date.now();
      const totalTime = ((endTime - startTime) / 1000).toFixed(2);

      console.log("--------------------------------");
      console.log(`Nice Work! You finished in ${totalTime} seconds!`);
      rl.close();
    }
  }

  askQuestion();
});
